sphinx/domains/std.py

```diff
@@ -223,12 +228,22 @@ def add_target_and_index(self, firstname: str, sig: str, signode: desc_signature
             node_id = make_id(self.env, self.state.document, prefix, optname)
             signode['ids'].append(node_id)
 
+            if 'compat-id' in self.options:
```

Comment:
Please add an old-styled ID to the signode always. Any new option is not needed to do that.